packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -761,10 +766,7 @@ export const DatePickerMixin = (subclass) =>
         this._applyInputValue(selectedDate);
       }
 
-      if (value !== this.value) {
-        this.validate();
```

Comment:
**note:** The validation is now triggered in `__dispatchChange()`.

The reason validation can be removed from this observer completely is because:

1. When the value property is changed programmatically, validation is already handled by the value property observer.
2. When the selected date change is triggered by the user, triggering validation is the responsibility of `__dispatchChange`.